main.tf

```diff
@@ -221,6 +248,30 @@ resource "aws_codebuild_project" "default" {
     location = var.artifact_location
   }
 
+  # Since the output type is restricted to S3 by the provider (this appears to
+  # be an bug in AWS, rather than an architectural decision; see this issue for
+  # discussion: https://github.com/hashicorp/terraform-provider-aws/pull/9652),
+  # this cannot be a CodePipeline output. Otherwise, _all_ of the artifacts
+  # would need to be secondary if there were more than one. For reference, see
+  # https://docs.aws.amazon.com/codepipeline/latest/userguide/action-reference-CodeBuild.html#action-reference-CodeBuild-config.
+  dynamic "secondary_artifacts" {
+    for_each = signum(length(var.secondary_artifact_location)) == 1 ? [""] : []
```

Comment:
Can you update this and the above conditional to be the more standard: `var.secondary_artifact_location == null ? [1] : []`?